### ISSUE TITLE
Ability to cycle through all records in a dupSort database

### ISSUE THREAD

**Issue by nwbrad** [Venemo/node-lmdb#110]
I deleted my original comments, because I determined what the issue is.

If you have an integer key of 0 (zero), the standard cursor loop will fail because the key returned is 0. 
From the example code: **for (var found = cursor.goToFirst(); found; found = cursor.goToNext()) {**

I read through the source... and I think the reason it occurs is that when MDB_NOTFOUND is encountered, it returns NULL, which will be evaluated to false in the cursor loop above. the same thing occurs when a integer key is equal to 0.  This is not specific to duplicates.  I could replicate it with any cursor that encounters a key of 0. Since I believe the source is correct, it the javascript examples and documentation that need to reflect this.

If I use a cursor loop as is shown below, then the 0 key works correctly in a loop.
**for (let found = cursor.goToFirst(); found !== null; found = cursor.goToNext()) {**

Thanks for your work on this.

**Comment by Venemo** [comment]
@nwbrad Wow, nice catch. With this change, were you able to cycle through that dupSort database correctly?

**Comment by nwbrad** [comment]
I was able to cycle through the dupSort database I have and it displayed everything as I expected with the null check.  I used !== since it checks the value and type, although I don't think there is a type for null.  != will probably work as well.

I can't remember the error I received on goToFirstDup or how I set it up.  It was just one of the many cases I was trying to see if I could figure out the problem.  I may have even been using it wrong.  Sorry.  

Thank you for updating the examples and readme.
